src/Flex.php

```diff
@@ -337,6 +331,27 @@ private function formatOrigin(string $origin): string
         return sprintf('<info>%s</> (<comment>%s</>): From %s', $matches[1], $matches[2], 'auto-generated recipe' === $matches[3] ? '<comment>'.$matches[3].'</>' : $matches[3]);
     }
 
+    private function shouldRecordOperation(OperationInterface $operation): bool
+    {
+        if ($operation instanceof UpdateOperation) {
+            $package = $operation->getTargetPackage();
+        } else {
+            $package = $operation->getPackage();
+        }
+
+        // FIXME: getNames() can return n names
+        $name = $package->getNames()[0];
```

Comment:
Is there a particular reason for this method to return more than one name? 🤔

Reply:
yes: `replace` and `provide` are giving more names to the package.